packages/realm/src/app-services/SyncSession.ts

```diff
@@ -441,12 +472,17 @@ export class SyncSession {
    * Can be either:
    *  - `reportIndefinitely` - the registration will stay active until the callback is unregistered
    *  - `forCurrentlyOutstandingWork` - the registration will be active until only the currently transferable bytes are synced
-   * @param callback - Called with the following arguments:
-   * 1. `transferred`: The current number of bytes already transferred
-   * 2. `transferable`: The total number of transferable bytes (the number of bytes already transferred plus the number of bytes pending transfer)
+   * @param callback - see {@link DynamicProgressNotificationCallback} and {@link PartitionBasedSyncProgressNotificationCallback}
```

Comment:
Perhaps something similar to this could suffice (mainly because the type itself exists by the parameter; so similarly to e.g. `mode`, we don't refer to the `ProgressMode`s available in the doc description here 👍 ).
```suggestion
   * @param callback - The function to call when the progress is updated.
```